src/cpp/session/modules/SessionRCompletions.R

```diff
@@ -157,80 +284,27 @@ assign(x = ".rs.acCompletionTypes",
    }
    
    tag <- sub(".*(?=@)", '', token, perl = TRUE)
+   metadata <- .rs.roxygenTagMetadata()
    
-   # All known Roxygen2 tags, in alphabetical order
-   #
-   # library(roxygen2)
-   # methods <- as.vector(methods("roxy_tag_parse"))
-   # prefix <- "roxy_tag_parse.roxy_tag_"
-   # paste0("@", gsub(prefix, "", methods[grepl(prefix, methods, fixed = TRUE)], fixed = TRUE), " ")
-   tags <- c(
-      "@aliases ",
-      "@author ",
-      "@backref ",
-      "@concept ",
-      "@describeIn ",
-      "@description ",
-      "@details ",
-      "@docType ",
-      "@encoding ",
-      "@eval ",
-      "@evalNamespace ",
-      "@evalRd ",
-      "@example ",
-      "@examples ",
-      "@examplesIf ",
-      "@export",
-      "@exportClass ",
-      "@exportMethod ",
-      "@exportPattern ",
-      "@exportS3Method ",
-      "@family ",
-      "@field ",
-      "@format ",
-      "@import ",
-      "@importClassesFrom ",
-      "@importFrom ",
-      "@importMethodsFrom ",
-      "@include ",
-      "@includeRmd ",
-      "@inherit ",
-      "@inheritDotParams ",
-      "@inheritParams ",
-      "@inheritSection ",
-      "@keywords ",
-      "@md",
-      "@method ",
-      "@name ",
-      "@note ",
-      "@noMd",
-      "@noRd",
-      "@order ",
-      "@param ",
-      "@rawNamespace ",
-      "@rawRd ",
-      "@rdname ",
-      "@references ",
-      "@return ",
-      "@returns ",
-      "@section ",
-      "@seealso ",
-      "@slot ",
-      "@source ",
-      "@template ",
-      "@templateVar ",
-      "@title ",
-      "@usage ",
-      "@useDynLib "
-   )
-   
-   matchingTags <- grep(paste("^", tag, sep = ""), tags, value = TRUE)
+   tags <- metadata$tags
+   recommend <- metadata$recommend
+   descriptions <- metadata$descriptions
+   vignette <- metadata$vignette
+
+   matching <- grepl(paste("^", tag, sep = ""), tags)
+
+   # put recommended tags first
+   keep <- c(which(matching & recommend), which(matching & !recommend))
    
    .rs.makeCompletions(tag,
-                       matchingTags,
-                       type = .rs.acCompletionTypes$ROXYGEN,
+                       results  = tags[keep],
+                       type     = .rs.acCompletionTypes$ROXYGEN,
+                       context  = .rs.acContextTypes$ROXYGEN, 
+                       meta     = descriptions[keep], 
```

Comment:
It is a bit quirky that this sends the `md` description for all the matched tags, and then later this `md` description is converted to html via `.rs.rpc.convert_roxygen_description()` which is just a call to `pandoc_convert()`.

Reply:
Maybe we should leave the pandoc_convert stuff out for now and then consider it later if we take the indexer approach I described in https://github.com/rstudio/rstudio/pull/12138#issuecomment-1279250007?

Reply:
Yeah, let's do this for this first pull request, it's already kind of big. Maybe we can pass the description through `.rs.htmlEscape()` at least, and reconsider `pandoc` to convert to html later, maybe when the `roxygen-tags.yml` start to actually use `md` syntax.